index.js

```diff
@@ -175,6 +187,28 @@ const parseArgs = (
   return result;
 };
 
+// Look ahead to next token in argv array:
+function peek(argv, pos) {
```

Comment:
CC: bakkot, @shadowspawn, @ljharb, a step towards a token based parser, as discussed in #52.

Benefit is it encapsulates the logic, and simplifies reading the codebase, e.g.,:

```js
  } else if (peek(argv, pos) !== TOKENS.FLAG) {
```